homeassistant/components/hassio/http.py

```diff
@@ -224,4 +224,10 @@ def should_compress(content_type: str) -> bool:
     """Return if we should compress a response."""
     if content_type.startswith("image/"):
         return "svg" in content_type
+    if content_type.startswith("application/"):
+        return (
+            "json" in content_type
+            or "xml" in content_type
+            or "javascript" in content_type
+        )
```

Comment:
Looks a bit fishy with the mimetype handling. Perhaps we should do.

```
full_type, _ , options = content_type.partition(";")
base_type, _, sub_type = fulltype.partition("/")
```

Avoids fully scanning string many times.

Reply:
I think we would need a complete mime type parser if we did that since we have places where is x-javascript, xhtml, etc